topics/migrating-3.md

```diff
@@ -407,6 +407,65 @@ To migrate, ensure `.bind()` is only called within a coroutine or suspending fun
 
 For more information on working with sockets, see the [Sockets documentation](server-sockets.md).
 
+### `PartData.FileItem.streamProvider()` is deprecated
+
+In previous versions of Ktor, the `.streamProvider()` function in `PartData.FileItem` was used to provide a file
+item's content as an `InputStream`. Starting with Ktor 3.0.0, this function has been deprecated.
+
+To migrate your application, replace `.streamProvider()` with the
+[`.provider()`](https://api.ktor.io/ktor-http/io.ktor.http.content/-part-data/-file-item/provider.html)
+function. The `.provider()` function returns a `ByteReadChannel`, which allows you to read data incrementally.
```

Comment:
> which allows you to read data incrementally

This is not what distinguishes it from the `InputStream` but the non-blocking behavior.